### PR TITLE
Extracted methods for shuffled test

### PR DESCRIPTION
**Pull request by unnonouno**
I extracted methods for shuffled test to make another shuffled test.

### CODE REVIEW

File: shuffle.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    # nccl is only supported on CUDA8
+    if 'centos6' in ret['base'] or \
+       ret['cuda'] == 'none' or \
+       ('ubuntu16' in ret['base'] and ret['cuda'] != 'cuda80'):
+        ret['nccl'] = 'none'
+
+    return ret
+
+
+def is_valid_version(version):
```
Comment (okuta) [inline shuffle.py]: This function is not called from anywhere.

File: shuffle.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    return bool(re.match(r'\d+(\.\d+)*', version))
```
Comment (okuta) [inline shuffle.py]: `re` is not imported.

File: shuffle.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+def get_shuffle_params(params, index):
+    random.seed(0)
+    keys = params.keys()
+    iters = [iter_shuffle(params[key]) for key in keys]
+    vals = next(itertools.islice(six.moves.zip(*iters), index, None))
+    ret = dict(zip(keys, vals))
+
+    # avoid SEGV
+    if ret['numpy'] == '1.9' and ret.get('h5py', 'none') != 'none':
```
Comment (okuta) [inline shuffle.py]: Why do you use `'none'` instead of `None`?

**Comment by unnonouno** [comment]
updated

File: shuffle.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    vals = next(itertools.islice(six.moves.zip(*iters), index, None))
+    ret = dict(zip(keys, vals))
+
+    # avoid SEGV
+    if ret['numpy'] == '1.9' and ret.get('h5py'):
+        ret['numpy'] = '1.10'
+
+    # nccl is only supported on CUDA8
+    if 'centos6' in ret['base'] or \
+       ret['cuda'] == 'none' or \
```
Comment (okuta) [inline shuffle.py]: Please fix.

File: shuffle.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+       ('ubuntu16' in ret['base'] and ret['cuda'] != 'cuda80'):
+        ret['nccl'] = 'none'
```
Comment (okuta) [inline shuffle.py]: ditto

File: shuffle.py (hunk 4 above)
Comment (okuta) [inline shuffle.py]: `argparse` returns `'none'`.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**